nevergrad/optimization/optimizerlib.py

```diff
@@ -1534,6 +1540,220 @@ def __init__(
 BO = ParametrizedBO().set_name("BO", register=True)
 
 
+class _HyperOpt(base.Optimizer):
+    # pylint: disable=too-many-instance-attributes
+    def __init__(self, parametrization: IntOrParameter,
+                 budget: tp.Optional[int] = None,
+                 num_workers: int = 1,
+                 *,
+                 prior_weight: float = 1.0,
+                 n_startup_jobs: int = 20,
+                 n_EI_candidates: int = 24,
+                 gamma: float = 0.25,
+                 verbose: bool = False) -> None:
+        super().__init__(parametrization, budget=budget, num_workers=num_workers)
+        try:
+            # try to convert parametrization to hyperopt search space
+            if not isinstance(parametrization, p.Instrumentation): raise NotImplementedError
+            self.space = self._get_search_space(parametrization.name, parametrization)
+            self._transform = None
+        except NotImplementedError:
+            self._transform = transforms.ArctanBound(0, 1)
+            self.space = {f"x_{i}": hp.uniform(f"x_{i}", 0, 1) for i in range(self.dimension)}
+
+        self.trials = Trials()
+        self.domain = Domain(fn=None, expr=self.space, pass_expr_memo_ctrl=False)
+        self.tpe_args = {
+            "prior_weight": prior_weight,
+            "n_startup_jobs": n_startup_jobs,
+            "n_EI_candidates": n_EI_candidates,
+            "gamma": gamma,
+            "verbose": verbose
+        }
+
+    def _get_search_space(self, param_name, param):
+        if isinstance(param, p.Instrumentation):
+            space = {}
+            space["args"] = {
+                    str(idx_param): self._get_search_space(str(idx_param), param[0][idx_param]) # type: ignore
+                    for idx_param in range(len(param[0].value))
+            }
+            space["kwargs"] = {
+                    param_name: self._get_search_space(param_name, param[1][param_name]) # type: ignore
+                    for param_name in param[1].value.keys()
+            }
+            return space
+        elif isinstance(param, (p.Log, p.Scalar)):
+            if (param.bounds[0][0] is None) or (param.bounds[1][0] is None):
+                raise ValueError(f"Scalar {param_name} not bounded.")
```

Comment:
Maybe replace  << raise ValueError(f"Scalar {param_name} not bounded.") >> by
                << return hp.lognormal(label=param_name, 0., 1.) >> ?

Reply:
I don't have strong opinion on that. I replaced it (but it will only make sense for real p.Scalar). 
Anyway, `hyperopt` is most suited to hyperparameter tuning and for such problems variables are usually bounded.